examples/jsm/nodes/core/NodeUtils.js

```diff
@@ -137,6 +137,12 @@ export function getValueFromType( type, ...params ) {
 
 	const last4 = type ? type.slice( - 4 ) : undefined;
 
+	if ( ( last4 === 'vec2' || last4 === 'vec3' || last4 === 'vec4' ) && params.length === 1 ) { // ensure same behaviour as in NodeBuilder.format()
```

Comment:
I think this actually might be even something we would like to do in core's Vector2, Vector3, and Vector4... I.e. so that, for example, new Vector2( 1 ) would produce new Vector2( 1, 1 ) instead of new Vector2( 1, 0 ).

Reply:
Interesting, I think it's cool, my only fear that it might break legacy projects.

Reply:
I think that breaking changes would be only in cases that aren't really widely used:
* `new Vector2()` -- was (0, 0), remains (0, 0), nothing changes,
* `new Vector2( x )` -- was (x, 0), becomes (x, x), **changes**,
* `new Vector2( x, y )` -- was (x, y), remains (x, y), nothing changes,
* `new Vector3()` -- was (0, 0, 0), remains (0, 0, 0), nothing changes,
* `new Vector3( x )` -- was (x, 0, 0), becomes (x, x, x), **changes**,
* `new Vector3( x, y )` -- was (x, y, 0), remains (x, y, 0), nothing changes,
* `new Vector3( x, y, z )` -- was (x, y, z), remains (x, y, z), nothing changes,
* `new Vector4()` -- was (0, 0, 0, 0), remains (0, 0, 0, 0), nothing changes,
* `new Vector4( x )` -- was (x, 0, 0, 1), becomes (x, x, x, 1), **changes**,
* `new Vector4( x, y )` -- was (x, y, 0, 1), remains (x, y, 0, 1), nothing changes,
* `new Vector4( x, y, z )` -- was (x, y, z, 1), remains (x, y, z, 1), nothing changes,
* `new Vector4( x, y, z, w )` -- was (x, y, z, w), remains (x, y, z, w), nothing changes.
So really nothing changes from the current version except `new Vector2/3/4( x )`.

@mrdoob @Mugen87 What do you think? Could this be done in core?